packages/compiler/src/server/tspconfig/completion.ts

```diff
@@ -80,6 +108,74 @@ export async function provideTspconfigCompletionItems(
         itemsFromEmitter.push(...more);
       }
       return [...itemsFromBuiltIn, ...itemsFromEmitter];
+    } else if (nodePath.length > CONFIG_PATH_LENGTH_FOR_LINTER_LIST && nodePath[0] === "linter") {
+      const linterName = nodePath[CONFIG_PATH_LENGTH_FOR_LINTER_LIST - 1];
+      const items: CompletionItem[] = [];
+      const linters = await linterProvider.listLibraries(tspConfigFile);
+
+      for (const [name, pkg] of Object.entries(linters)) {
+        if (linterName === "extends") {
+          // If a ruleSet exists for the linter, add it to the end of the library name.
+          const exports = await pkg.getModuleExports();
+          let additionalContent: string = "";
+          if (exports?.$linter?.ruleSets !== undefined) {
+            additionalContent = Object.keys(exports?.$linter?.ruleSets)[0];
+          }
+
+          const labelName = additionalContent.length === 0 ? name : `${name}/${additionalContent}`;
+          if (siblings.includes(labelName)) {
+            continue;
+          }
+
+          const newText = getNewTextValue(sourceType, labelName);
+          const item: CompletionItem = {
+            label: labelName,
+            kind: CompletionItemKind.Field,
+            documentation: (await pkg.getPackageJsonData())?.description ?? `Linters from ${name}`,
+            textEdit: getNewTextAndPosition(newText, source, tspConfigPosition),
+          };
+          items.push(item);
+        } else {
+          // enable/disable rules
+          const linter = await linterProvider.getLibrary(tspConfigFile, name);
+          if (!linter) {
+            return [];
+          }
+
+          const exports = await linter.getModuleExports();
+
+          if (exports?.$linter?.rules !== undefined) {
+            const more = resolveCompleteItems(
```

Comment:
you should be able to provide the items directly here, right? why call resolveCompleteItems here?